forc/src/ops/forc_init.rs

```diff
@@ -60,14 +60,20 @@ pub fn init(command: InitCommand) -> Result<()> {
         ))
     }
 
-    let project_name = command.name.unwrap_or_else(|| {
-        project_dir
-            .file_stem()
-            .context("Failed to infer project name from directory name.")
-            .unwrap()
-            .to_string_lossy()
-            .to_string()
-    });
+    let project_name = {
+        if let Some(name) = command.name {
+            name
+        } else {
+            let dir = project_dir
+                .file_stem()
+                .context("Failed to infer project name from directory name.");
+
+            match dir {
+                Ok(d) => d.to_string_lossy().to_string(),
+                Err(e) => anyhow::bail!("Failed to infer get current directory name: '{}'", e),
+            }
+        }
+    };
```

Comment:
It looks like we have some double-up in the error message:
```
"Failed to infer get current directory name: 'Failed to infer project name from directory name.'),
```
Good point about the `Cow` btw! I forgot `to_string_lossy()` returns a `Cow` and not a `String`.

How about something like this?
```rust
let project_name = match command.name {
    Some(name) => name,
    None => {
        project_dir
            .file_stem()
            .context("Failed to infer project name from directory name.")?
            .to_string_lossy()
            .into_owned()
    }
};
```